### PR TITLE
Rek shipment intransit backend

### PR DESCRIPTION
**Pull request by kilbergr**
## Description

This PR changes Shipment's PickupDate to ActualPickupDate throughout the code, adds an In Transit state to Shipment, and implements the transition from Approved to In Transit on the backend.
It also makes the In Transit queue visible on the office side.
Includes end 2 end test update.

## Setup

Add any steps or code to run in this section to help others prepare to run your code:

```
make db_dev_reset && make db_dev_migrate && ./bin/generate-test-data -scenario=7
```
Go to http://tsplocal:3000/api/v1/docs#/shipments/transportShipment and try it out with a shipment that's in the Approved state.
Go to http://officelocal:3000/queues/hhg_in_transit and see that same shipment.

## Code Review Verification Steps

* [x] End to end tests pass (`make e2e_test`).
* [x] Any migrations/schema changes also update the diagram in docs/schema/dp3.sqs (see [Updating and changing the model](https://github.com/transcom/mymove/blob/master/docs/schema/README.md#updating-and-changing-the-model)
* [x] Request review from a member of a different team.
* [x] Have the Pivotal acceptance criteria been met for this change?

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/160400191) for the backend changes
* [Pivotal story](https://www.pivotaltracker.com/story/show/160394779) for the queue changes

## Screenshots

screen shot 2018-09-13 at 5 24 48 pm

### CODE REVIEW

File: pkg/handlers/publicapi/shipments.go [hunk 1]
```diff
@@ -19,7 +19,7 @@ func payloadForShipmentModel(s models.Shipment) *apimessages.Shipment {
 		ID: *handlers.FmtUUID(s.ID),
 		TrafficDistributionList:             payloadForTrafficDistributionListModel(s.TrafficDistributionList),
 		ServiceMember:                       payloadForServiceMemberModel(s.ServiceMember),
-		PickupDate:                          *handlers.FmtDateTimePtr(s.PickupDate),
+		ActualPickupDate:                    *handlers.FmtDateTimePtr(s.ActualPickupDate),
```
Comment (chrisgilmerproj) [inline pkg/handlers/publicapi/shipments.go]: I'm a super fan of making this more clear!

File: migrations/20180913211923_rename_pickup_date_on_shipment.up.fizz [hunk 2]
```diff
@@ -0,0 +1 @@
+rename_column("shipments", "pickup_date", "actual_pickup_date")
```
Comment (chrisgilmerproj) [inline migrations/20180913211923_rename_pickup_date_on_shipment.up.fizz]: 👍

File: src/scenes/TransportationServiceProvider/api.js [hunk 3]
```diff
@@ -5,6 +5,7 @@ import { formatPayload } from 'shared/utils';
 export async function RetrieveShipmentsForTSP(queueType) {
   const queueToStatus = {
     new: ['AWARDED'],
+    in_transit: ['IN_TRANSIT'],
```
Comment (chrisgilmerproj) [inline src/scenes/TransportationServiceProvider/api.js]: Yes!

File: pkg/models/shipment.go [hunk 4]
```diff
@@ -31,10 +31,12 @@ const (
 	ShipmentStatusACCEPTED ShipmentStatus = "ACCEPTED"
 	// ShipmentStatusAPPROVED captures enum value "APPROVED"
 	ShipmentStatusAPPROVED ShipmentStatus = "APPROVED"
+	// ShipmentStatusINTRANSIT captures enum value "INTRANSIT"
```
Comment (chrisgilmerproj) [inline pkg/models/shipment.go]: `enum value "IN_TRANSIT"`

I do think its weird to have values with underscores.  Should we just remove all the underscores from the enums?  Or is this ok?

Comment (kilbergr) [reply pkg/models/shipment.go]: We have them throughout (e.g. PAYMENT_REQUESTED, AVIATION_CADET).

**Pull request merged**